Sources/Noora/Validator/Core/ValidatableRule.swift

```diff
@@ -0,0 +1,18 @@
+/// A protocol that defines a validation rule.
+///
+/// The `ValidatableRule` protocol allows for defining custom validation rules that can be applied to an input of type `String`.
+/// The protocol requires a property `error` that describes the validation error when the rule fails,
+/// and a method `validate(input:)` to perform the validation and return whether the input is valid or not.
+public protocol ValidatableRule {
+    /// The error that will be returned if the validation fails.
+    ///
+    /// This property represents the error associated with this validation rule,
+    /// and it will be used when the validation does not pass.
+    var error: ValidatableError { get }
+
+    /// Validates the provided input and checks if it satisfies the validation rule.
+    ///
+    /// - Parameter input: The input to validate, of type `String`.
+    /// - Returns: A boolean indicating whether the input passes the validation (`true`) or not (`false`).
+    func validate(input: String) -> Bool
```

Comment:
What do you think about merging `var error: ValidatableError` into `func validate`?

```swift
func validate(input: String) throws(ValidatableError)
```

Such that throwing means validation fail, and therefore we don't need to return a boolean.

Reply:
Thanks to that, `ValidatableError` can also be constructed at runtime allowing the usage of the value in the error description:

```swift
"xxx" is larger than x characters
```

Reply:
Yes, it's a good idea, but there's a problem with this approach. Swift doesn’t allow something like this:

```swift
func validate(input: String) throws(ValidatableError)
```  

because Swift treats it as `any Error`, resulting in the error: `Thrown type 'any ValidatableError' does not conform to the 'Error' protocol`.